app/models/course.rb

```diff
@@ -41,9 +39,9 @@ def closed?
     current_invitation.blank? || current_invitation.expired?
   end
 
-  def generate_invitation!(expiration_date)
-    invitation = invitations.build expiration_date: expiration_date, course: self
-    invitation.save_and_notify!
+  def create_invitation_for(expiration_date)
```

Comment:
I think the `for` preposition sounds a little awkward here, since the given date is not _the recipient_ of the invitation.  Why don't `create_invitation`, `generate_invitation`, or just `invite!`?